nucliadb_utils/src/nucliadb_utils/encryption/__init__.py

```diff
@@ -0,0 +1,57 @@
+# Copyright (C) 2021 Bosutech XXI S.L.
+#
+# nucliadb is offered under the AGPL v3.0 and as commercial software.
+# For commercial licensing, contact us at info@nuclia.com.
+#
+# AGPL:
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+import asyncio
+from concurrent.futures import ThreadPoolExecutor
+
+import nacl.secret
+import nacl.utils
+
+
+class EndecryptorUtility:
+    """
+    Utility class for encryption and decryption of sensitive data using the nacl library.
+    It uses a thread pool to avoid blocking the event loop.
+    """
+
+    def __init__(self, key: bytes, max_thread_pool_size: int = 1):
+        self.box = nacl.secret.SecretBox(key)
+        self.executor = ThreadPoolExecutor(max_workers=max_thread_pool_size)
+
+    async def encrypt_text(self, message: str) -> str:
+        message_bytes = message.encode()
+        encrypted = await self.encrypt(message_bytes)
+        return encrypted.hex()
+
+    async def encrypt(self, message: bytes) -> nacl.utils.EncryptedMessage:
+        loop = asyncio.get_running_loop()
```

Comment:
Using a threadpool might be a bit overkill for encrypting short texts with a symmetric algorithm, but it's there and definitely won't hurt :)